homeassistant/components/braviatv/__init__.py

```diff
@@ -1,24 +1,47 @@
 """The Bravia TV component."""
+import asyncio
+from datetime import timedelta
+import logging
 
 from bravia_tv import BraviaRC
+from bravia_tv.braviarc import NoIPControl
 
-from homeassistant.const import CONF_HOST, CONF_MAC
+from homeassistant.components.media_player import DOMAIN as MEDIA_PLAYER_DOMAIN
+from homeassistant.components.remote import DOMAIN as REMOTE_DOMAIN
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_PIN
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-from .const import BRAVIARC, DOMAIN, UNDO_UPDATE_LISTENER
+from .const import (
+    BRAVIA_CLIENT,
+    CLIENTID_PREFIX,
+    CONF_IGNORED_SOURCES,
+    DOMAIN,
+    NICKNAME,
+    UNDO_UPDATE_LISTENER,
+)
 
-PLATFORMS = ["media_player"]
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [MEDIA_PLAYER_DOMAIN, REMOTE_DOMAIN]
+SCAN_INTERVAL = timedelta(seconds=10)
 
 
 async def async_setup_entry(hass, config_entry):
     """Set up a config entry."""
     host = config_entry.data[CONF_HOST]
     mac = config_entry.data[CONF_MAC]
+    pin = config_entry.data[CONF_PIN]
+    ignored_sources = config_entry.options.get(CONF_IGNORED_SOURCES, [])
 
+    client = BraviaTVClient(hass, host, mac, pin, ignored_sources)
     undo_listener = config_entry.add_update_listener(update_listener)
 
+    await client.async_refresh()
```

Comment:
We use use the first refresh helper.
```suggestion
    await coordinator.async_config_entry_first_refresh()
```